internal/transport/publising.go

```diff
@@ -53,17 +55,21 @@ func PublishConnectionStatus(t Transport, dispatchers map[string]map[string]stri
 		},
 	}
 
-	err = t.SendControl(msg)
+	_, err = t.SendControl(msg)
 	if err != nil {
 		log.Error(err)
 	}
 	log.Debugf("published message %v to control topic", msg.MessageID)
 	log.Tracef("message: %+v", msg)
 }
 
-func PublishReceivedData(transport Transport, c <-chan yggdrasil.Data) {
+func PublishReceivedData(transport Transport, c <-chan yggdrasil.Data, e chan *pb.APIResponse) {
 	for d := range c {
-		err := transport.SendData(d)
+		data, err := transport.SendData(d)
+		if data != nil {
```

Comment:
this condition reduces the response for the http transport only, right ? since MQTT.SendData is expected to return nil.
Could this be encapsulated as part of the http.SendData if this behavior reserved for a specific type of transport ?